olive/model/handler/mixin/hf_config.py

```diff
@@ -62,9 +62,9 @@ def _get_hf_model_tokenizer(self):
         if self.hf_config is None:
             raise ValueError("HF model_config is not available")
 
-        return get_hf_model_tokenizer(
-            self._get_model_path_or_name(), **self.hf_config.get_loading_args_from_pretrained()
-        )
+        # don't provide loading args for tokenizer since it tries to serialize all kwargs
+        # TODO(anyone): only provide relevant kwargs, no use case for now to provide kwargs
+        return get_hf_model_tokenizer(self._get_model_path_or_name())
```

Comment:
Instead of removing the parameters here, we should pass kwargs to this function and not pass any parameters from call in `save_metadata_for_token_generation`. That way the API will stay generic and allow passing parameters in future without any change.

Reply:
I changed this here to keep the signature consistent with other similar methods. Since, it is a private method, I think it's okay? 

If we need to pass parameters in the future, we will probably have to handle `get_loading_args_from_pretrained` somewhere anyways.

Reply:
kwargs are optional and keyed so that wouldn't have changed the API. Anyways, there is work in future one way or other.

Reply:
sure! pushed a commit to accept kwargs